File: app/Models/Sport.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Sport extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'icon',
        'description',
    ];

    // One sport has many courts
    public function courts()
    {
        return $this->hasMany(Court::class);
    }
}

File: lib/features/sports/data/models/sport.dart
class Sport {
  final int id;
  final String name;
  final String? icon;
  final String? description;

  Sport({
    required this.id,
    required this.name,
    this.icon,
    this.description,
  });

  factory Sport.fromJson(Map<String, dynamic> json) {
    return Sport(
      id: json['id'],
      name: json['name'],
      icon: json['icon'],
      description: json['description'],
    );
  }
}